lib/os/mem_blocks.c

```diff
@@ -57,6 +57,45 @@ static int free_bits(sys_mem_blocks_t *mem_block, void *ptr, size_t bits)
 	return ret;
 }
 
+int sys_mem_blocks_alloc_continuous(sys_mem_blocks_t *mem_block, size_t count,
+				   void **out_block)
+{
+	int ret = 0;
+
+	if ((mem_block == NULL) || (out_block == NULL)) {
```

Comment:
Same deal: would prefer defense against "obvious bugs in the calling code" to be assertions and not runtime code bloat.